### ISSUE TITLE
Remove use of `NullThrownError` in language/nnbd/syntax/null_assertion_ambiguous_test.dart

### ISSUE THREAD

**Issue by nshahan** [dart-lang/sdk#50720]
The deprecated `NullThrownError` is being removed in Dart 3.0 in favor of a `TypeError`.

@eernstg 
In this case we can't simply change the use to a `TypeError` because that is what is thrown from the `!` operator. Any thoughts on how we should update this test to avoid the use of `NullThrownError`?
https://github.com/dart-lang/sdk/blob/022ca53db3dd82b99ed82d0f5c9f41f48cc2787f/tests/language/nnbd/syntax/null_assertion_ambiguous_test.dart#L23-L25

**Comment by eernstg** [comment]
The comment on line 18-22 indicates that this is actually intended to verify a syntactic property: `throw a!` is parsed as `throw (a!)` rather than `(throw a)!`.

However, `throw a` should be a compile-time [error](https://github.com/dart-lang/language/blob/master/accepted/2.12/nnbd/feature-specification.md#:~:text=It%20is%20an%20error%20if%20the%20static%20type%20of%20e%20in%20the%20expression%20throw%20e%20is%20not%20assignable%20to%20Object.) because the type of `a` is not assignable to `Object`. Moreover, the type of `throw ...` is `Never`, and that is a non-nullable type, so `(throw a)!` should cause a [warning](https://github.com/dart-lang/language/blob/master/accepted/2.12/nnbd/feature-specification.md#:~:text=It%20is%20a%20warning%20to%20use%20the%20null%20check%20operator%20(!)%20on%20an%20expression%20of%20type%20T%20if%20T%20is%20strictly%20non%2Dnullable%20.) (strictly non-nullable is just non-nullable with sound null safety).

In other words, we don't even need to worry about the run-time behavior in order to know that the parse was `throw (a!)`.

Perhaps `Expect.throws(() { throw a!; });` would do just fine.

**Comment by nshahan** [comment]
Fixed by https://dart-review.googlesource.com/c/sdk/+/275786.
